loopdb/sqlite_store.go

```diff
@@ -0,0 +1,739 @@
+// LoopIn struct.
+func (s *BaseDB) convertLoopInRow(row sqlc.GetLoopInSwapsRow,
+	updates []sqlc.SwapUpdate) (*LoopIn, error) {
+
+	htlcKeys, err := fetchHtlcKeys(
+		row.SenderScriptPubkey, row.ReceiverScriptPubkey,
+		row.SenderInternalPubkey, row.ReceiverInternalPubkey,
+		row.ClientKeyFamily, row.ClientKeyIndex,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	preimage, err := lntypes.MakePreimage(row.Preimage)
+	if err != nil {
+		return nil, err
+	}
+
+	swapHash, err := lntypes.MakeHash(row.SwapHash)
+	if err != nil {
+		return nil, err
+	}
+
+	loopIn := &LoopIn{
+		Contract: &LoopInContract{
+			SwapContract: SwapContract{
+				Preimage:         preimage,
+				AmountRequested:  btcutil.Amount(row.AmountRequested),
+				HtlcKeys:         htlcKeys,
+				CltvExpiry:       int32(row.CltvExpiry),
+				MaxSwapFee:       btcutil.Amount(row.MaxSwapFee),
+				MaxMinerFee:      btcutil.Amount(row.MaxMinerFee),
+				InitiationHeight: int32(row.InitiationHeight),
+				InitiationTime:   row.InitiationTime,
+				Label:            row.Label,
+				ProtocolVersion:  ProtocolVersion(row.ProtocolVersion),
+			},
+			HtlcConfTarget: int32(row.HtlcConfTarget),
+			ExternalHtlc:   row.ExternalHtlc,
+		},
+		Loop: Loop{
+			Hash: swapHash,
+		},
+	}
+
+	if row.LastHop != nil {
+		lastHop, err := route.NewVertexFromBytes(row.LastHop)
+		if err != nil {
+			return nil, err
+		}
+
+		loopIn.Contract.LastHop = &lastHop
+	}
+
+	// If we don't have any updates yet we can return early
+	if len(updates) == 0 {
+		return loopIn, nil
+	}
+
+	events, err := getSwapEvents(updates)
+	if err != nil {
+		return nil, err
+	}
+
+	loopIn.Events = events
+
+	return loopIn, nil
+}
+
+// getSwapEvents returns a slice of LoopEvents for the swap.
+func getSwapEvents(updates []sqlc.SwapUpdate) ([]*LoopEvent, error) {
+
+	events := make([]*LoopEvent, len(updates))
+
+	for i := 0; i < len(events); i++ {
+
+		events[i] = &LoopEvent{
+			SwapStateData: SwapStateData{
+				State: SwapState(updates[i].UpdateState),
+				Cost: SwapCost{
+					Server:   btcutil.Amount(updates[i].ServerCost),
+					Onchain:  btcutil.Amount(updates[i].OnchainCost),
+					Offchain: btcutil.Amount(updates[i].OffchainCost),
+				},
+			},
+			Time: updates[i].UpdateTimestamp.UTC(),
+		}
+
+		if updates[i].HtlcTxhash != "" {
+			chainHash, err := chainhash.NewHashFromStr(updates[i].HtlcTxhash)
+			if err != nil {
+				return nil, err
+			}
+
+			events[i].HtlcTxHash = chainHash
+		}
+	}
+
+	return events, nil
+}
+
+// convertOutgoingChanSet converts a comma separated string of channel IDs into
+// a ChannelSet.
+func convertOutgoingChanSet(outgoingChanSet string) (ChannelSet, error) {
+	// Split the string into a slice of strings
+	chanStrings := strings.Split(outgoingChanSet, ",")
+	channels := make([]uint64, len(chanStrings))
+
+	// Iterate over the chanStrings slice and convert each string to ChannelID
+	for i, chanString := range chanStrings {
+		chanID, err := strconv.Atoi(chanString)
```

Comment:
I think we want to use `strconv.ParseInt(chanString, 10, 64)` since `Atoi` returns just an `int`.